queue_entry.py

```diff
@@ -655,7 +653,8 @@ def execute(self):
         kappa = self._data_model.get_kappa()
         phi = self._data_model.get_kappa_phi()
 
-        self.diffractometer_hwobj.moveMotors({"kappa": kappa, "kappa_phi":phi})
+        if hasattr(self.diffractometer_hwobj, "in_kappa_mode") and self.diffractometer_hwobj.in_kappa_mode():
```

Comment:
A yes, you are right. `Not in_plate_mode` could have some more meaning than kappa.
Anyway, could you add `in_kappa_mode` to the GenericDiffractometer?